openscm/adapters/dice.py

```diff
@@ -0,0 +1,303 @@
+            default, unit = v
+            setattr(
+                self,
+                "_{}".format(n),
+                self._parameterset.get_scalar_view(("DICE", n), (), unit).get(default),
+            )
+
+        # Carbon cycle transition matrix
+        self._b11 = 1 - self._b12
+        self._b21 = self._b12 * self._M_atm_eq / self._M_u_eq
+        self._b22 = 1 - self._b21 - self._b23
+        self._b32 = self._b23 * self._M_u_eq / self._M_l_eq
+        self._b33 = 1 - self._b32
+
+        # Total CO2 emissions (GtCO2 per year)
+        self._E = self._parameterset.get_writable_timeseries_view(
+            ("Emissions", "CO2"), (), "GtCO2/a", self._start_time, self._period_length
+        )
+
+        # Concentration in atmosphere (GtC)
+        self._M_atm = self._parameterset.get_writable_timeseries_view(
+            ("Concentration", "Atmosphere"),
+            (),
+            "GtC",
+            self._start_time,
+            self._period_length,
+        )
+        self._M_atm.set(0, self._M_atm0)
+
+        # Carbon concentration increase in lower oceans (GtC from 1750)
+        self._M_l = self._parameterset.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "lower"),
+            (),
+            "GtC",
+            self._start_time,
+            self._period_length,
+        )
+        self._M_l.set(0, self._M_l0)
+
+        # Carbon concentration increase in shallow oceans (GtC from 1750)
+        self._M_u = self._parameterset.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "shallow"),
+            (),
+            "GtC",
+            self._start_time,
+            self._period_length,
+        )
+        self._M_u.set(0, self._M_u0)
+
+        # Increase in temperature of lower oceans (°C from 1900)
+        self._T_ocean = self._parameterset.get_writable_timeseries_view(
+            ("Temperature Increase", "Ocean", "lower"),
+            (),
+            "°C",
+            self._start_time,
+            self._period_length,
+        )
+        self._T_ocean.set(0, self._T_ocean0)
+
+        # Increase in radiative forcing (watts per m2 from 1900)
+        self._force = self._parameterset.get_writable_timeseries_view(
+            ("Radiative forcing",), (), "W/m^2", self._start_time, self._period_length
+        )
+
+        # Increase temperature of atmosphere (°C from 1900)
+        self._T_atm = self._parameterset.get_writable_timeseries_view(
+            ("Temperature Increase", "Atmosphere"),
+            (),
+            "°C",
+            self._start_time,
+            self._period_length,
+        )
+
+    def run(self) -> None:
+        """
+        Run the model over the full time range.
+        """
+        self._timestep = 0
+        for _ in range(self._timestep_count):
+            self.step()
+
+    def step(self) -> None:
+        """
+        Do a single time step.
+        """
+
+        self._timestep += 1
+
+        # Concentration in atmosphere (GtC)
```

Comment:
```suggestion
        # Atmospheric pool size
```

Reply:
These are the comments from the original DICE source. Shall we stick to these, though "concentration" is not correct here?

Reply:
I'd prefer to copy the DICE notation, but fix the comments to actually represent what they are (I'm not a fan of comments which are wrong but am ok with using misleading notation to maintain consistency)

Reply:
Cool, will mark them